src/Formik.tsx

```diff
@@ -332,7 +329,17 @@ export class Formik<ExtraProps = {}, Values = object> extends React.Component<
   };
 
   executeSubmit = () => {
-    this.props.onSubmit(this.state.values, this.getFormikActions());
+    const schema = this.getValidationSchema();
+    const { values } = this.state;
+    const actualizedValues = schema ? schema.cast(values) : values;
```

Comment:
This is breaking change

Reply:
Can you describe when this is used and add tests for when cast() is called?

Reply:
Why it's breaking change?
If you use validation schema so you've expected that transformation will work on submit, but if you don't so, nothing change.
And even so, v1 version, I think admit this.

I have a bunch of inputs described with yup schemas, and a lot of them contains `transform` and `test` yup method. For example, user input amount in dollars, but server expect to get this field in cents. Simple one line of code, and you get it for all schemas you need.

As for me, I've migrate to `formik` from `react-formal` and as a result I have a lot of yup schemas, but some of them doesn't work as expected, because of `test` and `transform functions` and some others bugs related to yup usage within formik.

Reply:
@jaredpalmer Have you ever tried to test real yup schema and not fake object?

Reply:
`const ValidationForm = FormFactory({ validationSchema: { validate } });`
This is a pice of code from here  https://github.com/jaredpalmer/formik/blob/master/test/withFormik.test.tsx#L179
And all others `validationSchema` has the same form, just `validate` custom method and that's all.

But when I try to add new test like this
```
const validationSchema = yup.object().shape({
        opensource: yup.string()
            .transform(function(value: string) {
                return value
                    ? value.toUpperCase()
                    : value;
            })
    });

const FormWithInput = () => (
      <Formik
        onSubmit={onSubmit}
        initialValues={{ opensource: 'yay' }}
        validationSchema={validationSchema}
      >
```

I get a warning, about uncaught Promise exception.
I've reviewed all tests, and didn't find ` yup.object()` as validationSchema, just fake `{ validation }`.

Reply:
Thank you for the PR !
This is highly desired, although in our usecase, we need to be able to validate multiple fields depending on each other as described [there](https://github.com/jquense/yup/issues/706)

It looks like to achieve this, we would need to pass the `options` argument to `cast`, unless you found a different approach, which I would be happy to use.

Cheers